core/server/data/migration/populate.js

```diff
@@ -30,20 +32,32 @@ populate = function populate(options) {
             context: {
                 internal: true
             }
-        },
-        tableSequence = Promise.mapSeries(schemaTables, function createTable(table) {
-            logger.info('Creating table: ' + table);
-            return commands.createTable(table);
-        });
+        };
 
     logger.info('Creating tables...');
 
-    if (tablesOnly) {
-        return tableSequence;
-    }
+    return new Promise(function populateDatabase(resolve, reject) {
+        db.knex.transaction(function populateDatabaseInTransaction(transaction) {
+            return Promise.mapSeries(schemaTables, function createTable(table) {
+                logger.info('Creating table: ' + table);
+                return commands.createTable(table, transaction);
+            }).then(function populateFixtures() {
+                if (tablesOnly) {
+                    return;
+                }
 
-    return tableSequence.then(function () {
-        return fixtures.populate(logger, modelOptions);
+                return fixtures.populate(logger, _.merge({}, {transacting: transaction}, modelOptions));
+            }).then(function commitTransaction() {
+                transaction.commit();
+                resolve();
+            }).catch(function rollbackTransaction(err) {
+                logger.warn('rolling back because of: ' + err.stack);
+
+                transaction.rollback();
+            });
+        }).catch(function populateDatabaseError() {
+            reject();
```

Comment:
The error handling here doesn't look correct to me. 

There's a double catch block. The inner one rolls back the transaction, but doesn't throw any error or reject the promise.
The outer one rejects a promise but without an error meaning there can be no other handling?